app/helpers/enterprises_helper.rb

```diff
@@ -87,4 +87,26 @@ def enterprise_url_selector(enterprise)
       main_app.producers_url
     end
   end
+
+  def hide_ofn_navigation?
+    # if we are not on a shopfront, a cart page, checkout page or the order confirmation page
+    # then we should show the OFN navigation
+    # whatever the current distributor has set for the hide_ofn_navigation preference
+    return false unless current_distributor && current_page?(main_app.enterprise_shop_path(current_distributor)) || # shopfront
+                        request.path.start_with?(main_app.checkout_path) || # checkout
+                        current_page?(main_app.cart_path) || # cart
+                        request.path.start_with?("/orders/") # order confirmation
+
+    distributor = if request.path.start_with?("/orders/")
+                    # if we are on an order confirmation page,
+                    # we need to get the distributor from the order, not the current one
+                    Spree::Order.find_by(number: params[:id]).distributor
```

Comment:
Although unlikely, it's possible that `find_by` could fail to find the order and return  `nil`. 
So we should probably use `find_by!` like [`Spree::OrdersController#show`](https://github.com/openfoodfoundation/openfoodnetwork/blob/master/app/controllers/spree/orders_controller.rb#L26-L28) 
```suggestion
                    Spree::Order.find_by!(number: params[:id]).distributor
```

But it feels like we're relying too much on internal knowledge of another class. What if we could update  `current_distributor` to provide the right one in this context? (that is, in `Spree::Core::ControllerHelpers::Order#spree_current_order`). 
🤔  Hmm I can't get my head around that, so I don't have any better ideas.

Reply:
Actually I think this is relevant for the whole method. If we could define in the controllers when to allow hiding of navigation, it would cleaner. Something like `before_action :allow_hiding_nav, only: :show`
But I have to admit I don't know how to do that right now..

Reply:
`current_distributor` is defined in `EnterprisesHelper`:
https://github.com/openfoodfoundation/openfoodnetwork/blob/45f6e6f0d53d5c0f387eb91dcfbd59f8eb13fc45/app/helpers/enterprises_helper.rb#L4-L6

but it's the distributor used in the nav bar (ie. the one with a current order, even empty associated): 
<img width="544" alt="Capture d’écran 2023-03-17 à 09 54 07" src="https://user-images.githubusercontent.com/296452/225857915-5df115c1-0035-4d88-99b1-7b31d6df2c36.png">

Which could be different that the distributor of the order we're currently seen in the order confirmation page. That's why i've created that `if`
```ruby
    distributor = if request.path.start_with?("/orders/")
                    # if we are on an order confirmation page,
                    # we need to get the distributor from the order, not the current one
                    Spree::Order.find_by(number: params[:id]).distributor
                  else
                    current_distributor
                  end
```
I didn't want to update that `EnterpriseHelper#current_distributor` method, since it's widely used and I was afraid of the consequences that could occur. 

(btw I've then added a special spec for this particular case: f9ff4605e5)

Reply:
I've thinking about your solution with that `before_action` for the concerned controllers, and I think my proposed solution is wayyyy better thanks to you: 5fb887ad83

What do you think?